File: incident-service/src/main/java/com/emergency/incident/dto/IncidentDTO.java
package com.emergency.incident.dto;

import com.emergency.incident.model.IncidentStatus;
import com.emergency.incident.model.IncidentType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class IncidentDTO {
    private Long id;
    private String citizenName;
    private String citizenPhone;
    private IncidentType incidentType;
    private Double latitude;
    private Double longitude;
    private String locationAddress;
    private String notes;
    private Long createdBy;
    private Long assignedUnitId;
    private String assignedUnitName;
    private String assignedUnitType;
    private IncidentStatus status;
    private LocalDateTime timestamp;
    private LocalDateTime dispatchedAt;
    private LocalDateTime resolvedAt;
}

File: backend/incident-service/src/dto/incident.dto.ts
import { IncidentStatus, IncidentType } from '../entities/Incident';
import { StationType } from '../entities/ResponderStation';

export interface CreateIncidentRequest {
  citizenName: string;
  citizenPhone: string;
  incidentType: IncidentType;
  latitude: number;
  longitude: number;
  locationAddress: string;
  notes?: string;
}

export interface UpdateStatusRequest {
  status: IncidentStatus;
}

export interface AssignUnitRequest {
  unitId: number;
  unitName: string;
  unitType: string;
}

export interface IncidentDTO {
  id: number;
  citizenName: string;
  citizenPhone: string;
  incidentType: IncidentType;
  latitude: number;
  longitude: number;
  locationAddress: string;
  notes?: string;
  status: IncidentStatus;
  assignedUnitId?: number;
  assignedUnitName?: string;
  assignedUnitType?: string;
  createdAt: Date;
  dispatchedAt?: Date;
  updatedAt: Date;
}

export interface CreateStationRequest {
  name: string;
  stationType: StationType;
  latitude: number;
  longitude: number;
  address: string;
  phoneNumber: string;
  capacity?: number;
}
